databroker/core.py

```diff
@@ -193,6 +196,21 @@ def __getstate__(self):
                               for k, arg in self._captured_init_kwargs.items()})
         return OrderedDict(cls=self.classname, args=args, kwargs=kwargs)
 
+    def get(self, **user_parameters):
+        if self._data_source is None:
+            """Instantiate the DataSource for the given parameters"""
+            plugin, open_args = self._create_open_args(user_parameters)
+            data_source = plugin(**open_args)
+            data_source.catalog_object = self._catalog
+            data_source.name = self.name
+            data_source.description = self._description
+            data_source.cat = self._catalog
+            self._data_source = data_source
+            return data_source
+        else:
+            _, open_args = self._create_open_args(user_parameters)
+            self._data_source.__init__(**open_args)
```

Comment:
If we are going to re-run `__init__` why not just start from scratch?

Reply:
Updated this, it doesn't work this way any longer.